packages/user-service/src/resolver.ts

```diff
@@ -0,0 +1,101 @@
+import { User } from './schema';
+import * as _ from 'lodash';
+import moment from 'moment';
+import { UserGroupAPIHelper } from './helpers';
+
+export const UserResolver = {
+  Query: {
+    getUser(root: any, args: any, ctx: any) {
+      return User.findOne({ uid: args.uid })
+        .then(data => data)
+        .catch(err => err);
+    },
+    listUsers(root: any, args: any, ctx: any) {
+      return User.find()
+        .then(data => data)
+        .catch(err => err);
+    },
+    userLogin(root: any, args: any, ctx: any) {
```

Comment:
This should be a mutation, as it is modifying the data.
Also when will this be used? Is it when a new user logs in to the portal, but that user is not in the db? If yes, then this resolver needs to be renamed. It is very misleading, as `userLogin` means that this will either allow the user to login/authenticate, or check if the user is logged in or not.
Rename it to `addUserfromLDAP` or something similar.

Reply:
Updated the code amd moved to the mutation section